libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -158,9 +158,30 @@ MAV_RESULT AP_Scripting::handle_command_int_packet(const mavlink_command_int_t &
     return MAV_RESULT_UNSUPPORTED;
 }
 
-bool AP_Scripting::repl_start(void) {
+bool AP_Scripting::repl_start(relp_type type) {
     if (terminal.session) { // it's already running, this is fine
-        return true;
+        if (type == terminal.type) {
+            return true;
+        } else {
+            // running but wrong type, stop current session, second request will start new
```

Comment:
I'm not sure if this should actually be able to force kill your existing session. I'd have been inclined to say that we don't stop here, and we just return false.